### ISSUE TITLE
Cache invalidation

### ISSUE THREAD

**Issue by Vilddjur** [cthit/chalmersit-rails#74]
We need to invalidate the cache when a post or comment is edited/updated. There is a fancy after_save method for this and I looked at doing something along the lines of what we are doing on [account](https://github.com/cthit/chalmersit-account-rails/blob/develop/app/models/ldap_group.rb#L96) but this wont work here sadly since we are not cacheing [objects](https://github.com/cthit/chalmersit-account-rails/blob/develop/app/models/ldap_group.rb#L27) but rather [fragments](https://github.com/cthit/chalmersit-rails/blob/develop/app/views/posts/index.html.erb#L3). I looked into doing something with [expire_fragment](http://api.rubyonrails.org/classes/ActionController/Caching/Fragments.html#method-i-expire_fragment) but I am not sure how to apply this to our way of cacheing since I haven't found any examples of that.

**Comment by lindskogen** [comment]
`app/views/home/index.html.erb`

```erb
Change
<% cache ['home_posts', I18n.locale] do  %>

to 
<% cache ['home_posts', I18n.locale, @posts.first] do  %>

will do the trick
```

Now the cached fragment depends on the newest post!

We still need `expire_fragment` for when posts are edited though. So it would be a great idea to look into it! Check out the railscast episode for more info http://railscasts.com/episodes/90-fragment-caching

**Comment by Edholm** [comment]
> cache_key generates a string based on the model’s id and updated_at attributes. This is a common convention and has the benefit of invalidating the cache whenever the product is updated. In general, when you use low-level caching for instance level information, you need to generate a cache key.

[Source](http://guides.rubyonrails.org/caching_with_rails.html#low-level-caching)

**Comment by lindskogen** [comment]
Yes, when you pass an ActiveRecord model to the cache method, it will by default call `.cache_key` on the model

**Comment by Vilddjur** [comment]
Solved in part by 0a0fb0c87eb34d6ee6a2b0b0b491d66b1788ed1f, will keep an eye out for more issues with caching.

**Comment by malmers** [comment]
Closed until further issue arises
